src/System.IO.Pipelines/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -7,41 +7,60 @@
 
 namespace System.IO.Pipelines
 {
-    internal class BufferSegment
+    public interface IMemoryList<T>
+    {
+        ReadOnlyMemory<T> Memory { get; }
```

Comment:
Then it should be called IReadOnlyMemoryList, but I thought we can use use R/W list (as we discussed)